src/common.c

```diff
@@ -585,6 +585,33 @@ void ofi_straddr_log_internal(const char *func, int line,
 	}
 }
 
+int ofi_is_any_addr(struct sockaddr *sa)
+{
+	struct in_addr ia_any = {
+		.s_addr = INADDR_ANY,
+	};
+	struct in6_addr ia6_any = IN6ADDR_ANY_INIT;
+	struct sockaddr_in *sa_in;
+	struct sockaddr_in6 *sa_in6;
+
+	if (!sa)
+		return 0;
+
+	switch(sa->sa_family) {
+	case AF_INET:
+		sa_in = (struct sockaddr_in *)sa;
+		return !memcmp(&sa_in->sin_addr.s_addr, &ia_any,
+				sizeof(ia_any)) && sa_in->sin_port ? 1 : 0;
+	case AF_INET6:
+		sa_in6 = (struct sockaddr_in6 *)sa;
+		return !memcmp(&sa_in6->sin6_addr, &ia6_any,
+				sizeof(ia6_any)) && sa_in6->sin6_port ? 1 : 0;
+	default:
+		FI_WARN(&core_prov, FI_LOG_CORE, "Unknown address format!\n");
+		return 0;
+	}
+}
```

Comment:
Typically 'any' address checks only look at the IP portion.  The port is ignored.  I recommend following the same here.  If port checks are needed, handle them in the location where the port check matters.

I would also have this function call subroutines.  ofi_ipv4_any_addr() and ofi_ipv6_any_addr().